src/cli.ts

```diff
@@ -166,19 +163,18 @@ async function runWithPaths(paths: Array<string>, options: CLIOptions, callback:
 
   async function processDirectory(path: string): Promise<void> {
     let children = await readdir(path);
-    pending.unshift(
-      ...children
-        .filter(child => {
-          if (options.modernizeJS) {
-            return child.endsWith('.js');
-          } else {
-            return child.endsWith('.coffee') ||
-              child.endsWith('.litcoffee') ||
-              child.endsWith('.coffee.md');
-          }
-        })
-        .map(child => join(path, child))
-    );
+
+    for (let child of children) {
```

Comment:
Looks like this isn't traversing subdirectories, as mentioned in #1131. With this refactor, it should be a lot easier to fix that, although I'm also fine fixing it as a follow-up to keep this change more behavior-preserving.

Reply:
Good point. I added that too since, as you said, it wasn't too hard now that it's all async/await.